### PR TITLE
blast species lozenges onclick function

### PR DESCRIPTION
**Pull request by ens-ma7**
## Type

- [ ] Bug fix
- [x] New feature
- [ ] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1481

## Importance
N/A

## Description
select species checkbox when clicking on the species lozenge in blast. Note that if the species checkbox is already selected/checked, nothing happen when clicking the species lozenge. Only action is select the species checkbox if it is not selected.

## Deployment URL
http://blast-specieslozenge.review.ensembl.org

## Views affected
Blast

### Other effects
N/A

- [ ] I have checked any requirements for Google Analytics
- [ ] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

## Possible complications
N/A

### CODE REVIEW

**Comment by ens-ma7** [comment]
Note, not sure if this needs a test to be written. there was no test for ToolsAppBar hence didnt create one but happy to do one if it makes sense to have one.

**Review by azangru — APPROVED** [review]
The line `updateSelectedSpecies({ isChecked: true, genomeId })` suggests that the name of the `isChecked` parameter of the `updateSelectedSpecies` was influenced by the UI of the species selector that had checkboxes. That's not very good. I'd rather we had two methods instead: `addSelectedSpecies` and `removeSelectedSpecies`, which just take a genome id.

On the other hand, we'll have to replace this provisional species selector with a different species selector anyway; so I am not super bothered by this.

**Comment by ens-ma7** [comment]
@azangru it also didnt feel right but wasnt sure about the approach but thanks for the hint and i think it makes sense to add these two functions because even though the species selector will be different one thing that i have noticed is the same is the presence of the checkbox and also to select the species using the checkbox. I will make the suggested change. And do you think we should write a test for this (clicking on the species lozenge will add a species to the state?)

**Review by jyothishnt — APPROVED** [review]

**Comment by azangru** [comment]
> do you think we should write a test for this (clicking on the species lozenge will add a species to the state?)

Won't hurt 🙂

File: src/content/app/tools/shared/components/tools-app-bar/ToolsAppBar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    store
+  };
+};
+
+describe('ToolsAppBar', () => {
+  describe('Species Lozenge click', () => {
+    it('updates the selectedSpecies state', () => {
+      const { container, store } = renderComponent();
+
+      const speciesLozenge = container.querySelector('.inUseInactive');
```
Comment (azangru) [inline src/content/app/tools/shared/components/tools-app-bar/ToolsAppBar.test.tsx:89]: `inUseInactive` is a strange class to use to target the species lozenge. We can use the class `.species`; or we can use the `getByText` helper from the same object from which we are pulling the `container` and  find an element using `getByText('Human')`.

Comment (ens-ma7) [reply src/content/app/tools/shared/components/tools-app-bar/ToolsAppBar.test.tsx]: Done.

**Review by azangru — APPROVED** [review]

**Pull request merged**